### PR TITLE
feat: add reasoning effort to agent config model + support in core

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

- add `reasoningEffort` on the `agent_configurations` table (optional)
- add support for `reasoning_effort` config on `core` LLM / Chat blocks for OAI provider (as `extras` config).

## Risk

critical path for pretty much everything + database migration

## Deploy Plan

1. apply DB schema migration (prodbox)
2. deploy `core` (adds support for the config) and deploy `front` (functionally a no-op)

### CODE REVIEW

File: front/migrations/db/migration_136.sql [hunk 1]
```diff
@@ -0,0 +1,26 @@
+-- -- This migration is dependant on a backfill script
+-- -- The backfill script is: 
+-- -- run psql with --set=backfilled=1 argument if you have rune the script.
+
+CREATE OR REPLACE FUNCTION perform_migration(backfilled boolean DEFAULT false)
+RETURNS VARCHAR AS $$
+BEGIN
+    IF NOT backfilled THEN
+        RAISE NOTICE 'The backfill script:  is required before applying this migation. If you already did it, run psql with --set=backfilled=1 argument.';
+    END IF;
```
Comment (tdraier) [inline front/migrations/db/migration_136.sql]: what backfill is needed for that ?

**Review by tdraier — APPROVED** [review]
ooc, what is reasoning effort ?

File: front/migrations/db/migration_136.sql (hunk 1 above)
Comment (fontanierh) [reply front/migrations/db/migration_136.sql]: Removed, for some reason our migration generator adds all that stuff (I never want it personally)

**Comment by fontanierh** [comment]
@tdraier it's the amount of test time compute you want to use for o1 models: https://platform.openai.com/docs/api-reference/chat/create

**Review by spolu — COMMENTED** [review]
LGTM

New agent config field! Show time!

**Pull request merged**